scaper/core.py

```diff
@@ -209,16 +210,45 @@ def generate_from_jams(jams_infile,
     sc.fade_in_len = ann.sandbox.scaper['fade_in_len']
     sc.fade_out_len = ann.sandbox.scaper['fade_out_len']
 
-    # Generate audio and save to disk
+    # Pull generation parameters from annotation
     reverb = ann.sandbox.scaper['reverb']
 
+    if 'fix_clipping' in ann.sandbox.scaper.keys():
+        fix_clipping = ann.sandbox.scaper['fix_clipping']
+    else:
+        fix_clipping = False
+
+    if 'peak_normalization' in ann.sandbox.scaper.keys():
+        peak_normalization = ann.sandbox.scaper['peak_normalization']
+    else:
+        peak_normalization = False
+
     # Cast ann.sandbox.scaper to a Sandbox object
     ann.sandbox.scaper = jams.Sandbox(**ann.sandbox.scaper)
-    soundscape_audio, event_audio_list = \
-        sc._generate_audio(audio_outfile, ann, reverb=reverb,
+
+    # Generate audio
+    soundscape_audio, event_audio_list, scale_factor, ref_db_change = \
+        sc._generate_audio(audio_outfile,
+                           ann,
+                           reverb=reverb,
+                           fix_clipping=fix_clipping,
+                           peak_normalization=peak_normalization,
                            save_isolated_events=save_isolated_events,
                            isolated_events_path=isolated_events_path,
                            disable_sox_warnings=disable_sox_warnings)
+    
+    # TODO: Stick to heavy handed overwriting for now, in the future we
+    # should consolidate this with what happens inside _instantiate().
+    ann.sandbox.scaper.reverb = reverb
+    ann.sandbox.scaper.fix_clipping = fix_clipping
+    ann.sandbox.scaper.peak_normalization = peak_normalization
+    ann.sandbox.scaper.save_isolated_events = save_isolated_events
+    ann.sandbox.scaper.isolated_events_path = isolated_events_path
+    ann.sandbox.scaper.disable_sox_warnings = disable_sox_warnings
+    ann.sandbox.scaper.peak_normalization_scale_factor = scale_factor
+    ann.sandbox.scaper.ref_db_change = ref_db_change
+    ann.sandbox.scaper.ref_db_generated = sc.ref_db + ref_db_change
```

Comment:
This code is duplicated between here and in `generate`. Might be worth refactoring at some point.

Reply:
Agreed - I'll create an issue for it so we address in a future PR

Reply:
https://github.com/justinsalamon/scaper/issues/135